3.evaluate_model/scripts/nbconverted/process_jump_phenotype_profiles.py

```diff
@@ -0,0 +1,283 @@
+    .reset_index(drop=True)
+)
+
+jump_focused_top_results_df.to_csv(top_results_summary_file, sep="\t", index=False)
+
+print(jump_focused_top_results_df.shape)
+jump_focused_top_results_df.head()
+
+
+# ## Summarize data
+
+# In[8]:
+
+
+# How many unique plates?
+jump_pred_df.Metadata_Plate.nunique()
+
+
+# In[9]:
+
+
+# How many different individual treatments?
+jump_pred_df.query("Metadata_model_type == 'final'").treatment_type.value_counts()
+
+
+# In[10]:
+
+
+# How many unique treatments per treatment type?
+jump_pred_df.groupby("treatment_type").treatment.nunique()
+
+
+# In[11]:
+
+
+# How many treatments with phenotype predictions?
+jump_pred_df.query("Metadata_model_type == 'final'").phenotype.value_counts()
+
+
+# ## Convert data to phenotypic profiles
+
+# In[12]:
+
+
+metadata_columns = [
+    "Metadata_Plate",
+    "treatment",
+    "treatment_type",
+    "Cell_type",
+    "Time",
+    "Metadata_Well",
+    "cell_count"
+]
+
+
+# In[13]:
+
+
+jump_wide_final_df = (
+    jump_pred_df
+    .query("Metadata_model_type == 'final'")
+    .drop(columns=["p_value"])
+    .pivot(index=metadata_columns, columns="phenotype", values="comparison_metric_value")
+    .reset_index()
+)
+
+jump_wide_final_df.to_csv(final_jump_phenotype_file, sep="\t", index=False)
+
+print(jump_wide_final_df.shape)
+jump_wide_final_df.head()
+
+
+# In[14]:
+
+
+jump_wide_shuffled_df = (
+    jump_pred_df
+    .query("Metadata_model_type == 'shuffled'")
+    .drop(columns=["p_value"])
+    .pivot(index=metadata_columns, columns="phenotype", values="comparison_metric_value")
+    .reset_index()
+)
+
+jump_wide_shuffled_df.to_csv(shuffled_jump_phenotype_file, sep="\t", index=False)
+
+print(jump_wide_shuffled_df.shape)
+jump_wide_shuffled_df.head()
+
+
+# ## Apply UMAP to phenotypic profiles
+
+# In[15]:
+
+
+umap_random_seed = 123
+umap_n_components = 2
+
+feature_columns = jump_wide_final_df.drop(columns=metadata_columns).columns.tolist()
+print(len(feature_columns))
+
+
+# In[16]:
+
+
+# Initialize UMAP
+umap_fit = umap.UMAP(random_state=umap_random_seed, n_components=umap_n_components)
+
+# Fit UMAP and convert to pandas DataFrame
+embeddings = pd.DataFrame(
+    umap_fit.fit_transform(jump_wide_final_df.loc[:, feature_columns]),
+    columns=[f"UMAP{x}" for x in range(0, umap_n_components)],
+)
+
+# Combine with metadata
+umap_with_metadata_df = pd.concat([jump_wide_final_df.loc[:, metadata_columns], embeddings], axis=1).assign(model_type="final")
+
+
+# In[17]:
+
+
+# Initialize UMAP
+umap_fit = umap.UMAP(random_state=umap_random_seed, n_components=umap_n_components)
+
+# Fit UMAP and convert to pandas DataFrame
+embeddings = pd.DataFrame(
+    umap_fit.fit_transform(jump_wide_shuffled_df.loc[:, feature_columns]),
+    columns=[f"UMAP{x}" for x in range(0, umap_n_components)],
+)
+
+# Combine with metadata
+umap_shuffled_with_metadata_df = pd.concat([jump_wide_shuffled_df.loc[:, metadata_columns], embeddings], axis=1).assign(model_type="shuffled")
```

Comment:
Could create a function here as well

Reply:
thanks! For this one I will create a function :) There is a lot of redundant code here which could easily be made into a function. Also, the inner workings of this function is less important to know about than with the pandas chaining example.